dkg-gadget/src/worker.rs

```diff
@@ -650,16 +487,73 @@ where
 		self.process_finished_rounds();
 	}
 
+	fn process_finished_rounds(&mut self) {
+		for finished_round in self.rounds.get_finished_rounds() {
+			self.handle_finished_round(finished_round);
+		}
+
+		if let Some(mut next_rounds) = self.next_rounds.take() {
```

Comment:
We might be able to remove this, I don't think we'll ever conduct any voting process with the `next_rounds` value